config/application.rb

```diff
@@ -31,5 +31,8 @@ class Application < Rails::Application
 
     # Custom error pages
     config.exceptions_app = self.routes
+
+    config.paths["config/initializers"] << AppConfig.initializers_root.to_s if AppConfig.load_external_config?
+    config.paths["app/views"].unshift AppConfig.config_root.join("views").to_s if AppConfig.load_external_config?
```

Comment:
Maybe `AppConfig.config_root.join("views")` should be a method on `AppConfig` (e.g., `AppConfig.views_root`).

Reply:
Or we get rid of initializers_root. I'm not sure what a separate configurable initializers root or app views root is useful for. In all my testing use cases, the only thing I needed to vary was the app config root itself, so instead of being in /etc/ it was in a directory in my home directory so I could do local testing between awesim and ondemand easily while in development mode.

Reply:
I am fine with either decision you make, but should probably be consistent between both `initializers` and `views`.

Although some pros for instance methods:

- if we ever do change the location of those paths it would be easier to support backwards compatibility (or if we wanted to check for the existence of multiple locations)
- we can move the logic involving `rails_env` to the Configuration object itself as:

  ```ruby
  def initializers_root
    Pathname.new( ... ) if load_external_config?
  end
  ```

  and the `config/application.rb` can be updated with

  ```ruby
  config.paths["config/initializers"].push(AppConfig.initializers_root.to_s) if AppConfig.initializers_root
  ```